gammapy/cube/psf_map.py

```diff
@@ -135,12 +156,25 @@ def geom(self):
     @classmethod
     def read(cls, filename, **kwargs):
         """Read a psf_map from file and create a PSFMap object"""
-        psfmap = Map.read(filename, **kwargs)
-        return cls(psfmap)
-
-    def write(self, *args, **kwargs):
+        with fits.open(filename, memmap=False) as hdulist:
+            psfmap = Map.from_hdulist(hdulist, "PSFMAP", "BANDSPSF", "auto")
+            if "EXPMAP" in hdulist:
+                expmap = Map.from_hdulist(hdulist, "EXPMAP", "BANDSEXP", "auto")
+            else:
+                expmap = None
+        return cls(psfmap, expmap)
+
+    def write(self, filename, overwrite=False):
         """Write the Map object containing the PSF Library map."""
-        self._psf_map.write(*args, **kwargs)
+        hdulist = self._psf_map.to_hdulist(hdu="PSFMAP", hdu_bands="BANDSPSF")
+        if self.exposure_map is not None:
+            new_hdulist = self._exposure_map.to_hdulist(
```

Comment:
Does `hdulist` have an extend method like list does?
Seems unnecessarily complex to make a list with two things and then take them out indiviaually again copy things over. I think hdulist also supports slicing, so if there's an empty primary HDU in the way `[1:]` should work?
The fewer times we repeat the keynames "EXPMAP" and "BANDSEXP", the more maintainable the code is.